src/Compilers/CSharp/Test/Semantic/Semantics/StaticNullChecking_TypeInference.cs

```diff
@@ -0,0 +1,1481 @@
+        [Fact(Skip = "TODO")]
+        public void LocalVar_NonNull()
+        {
+            var source =
+@"class C
+{
+    static void Main()
+    {
+        var s = string.Empty;
+        s.ToString();
+        s = null;
+        s.ToString();
+    }
+}";
+
+            var comp = CreateStandardCompilation(
+                source,
+                parseOptions: TestOptions.Regular8);
+            comp.VerifyDiagnostics(
+                // (7,13): warning CS8600: Cannot convert null to non-nullable reference.
+                //         s = null;
+                Diagnostic(ErrorCode.WRN_NullAsNonNullable, "null").WithLocation(7, 13));
+
+            var tree = comp.SyntaxTrees[0];
+            var model = comp.GetSemanticModel(tree);
+            var declarator = tree.GetRoot().DescendantNodes().OfType<VariableDeclaratorSyntax>().First();
+            var symbol = (LocalSymbol)model.GetDeclaredSymbol(declarator);
+            Assert.Equal("System.String", symbol.Type.ToTestDisplayString());
+            Assert.Equal(false, symbol.Type.IsNullable);
+        }
+
+        // PROTOTYPE(NullableReferenceTypes): `var s0 = b ? string.Empty : string.Empty;`
+        // should declare non-nullable string.
+        [Fact(Skip = "TODO")]
+        public void LocalVar_ConditionalOperator()
+        {
+            var source =
+@"class C
+{
+    static void F(bool b)
+    {
+        var s0 = b ? string.Empty : string.Empty;
+        var s1 = b ? string.Empty : null;
+        var s2 = b ? null : string.Empty;
+    }
+}";
+
+            var comp = CreateStandardCompilation(
+                source,
+                parseOptions: TestOptions.Regular8);
+            comp.VerifyDiagnostics();
+
+            var tree = comp.SyntaxTrees[0];
+            var model = comp.GetSemanticModel(tree);
+            var declarators = tree.GetRoot().DescendantNodes().OfType<VariableDeclaratorSyntax>().ToArray();
+
+            var symbol = (LocalSymbol)model.GetDeclaredSymbol(declarators[0]);
+            Assert.Equal("System.String", symbol.Type.ToTestDisplayString());
+            Assert.Equal(false, symbol.Type.IsNullable);
+
+            symbol = (LocalSymbol)model.GetDeclaredSymbol(declarators[1]);
+            Assert.Equal("System.String", symbol.Type.ToTestDisplayString());
+            Assert.Equal(true, symbol.Type.IsNullable);
+
+            symbol = (LocalSymbol)model.GetDeclaredSymbol(declarators[2]);
+            Assert.Equal("System.String", symbol.Type.ToTestDisplayString());
+            Assert.Equal(true, symbol.Type.IsNullable);
+        }
+
+        [Fact]
+        public void LocalVar_Array_01()
+        {
+            var source =
+@"class C
+{
+    static void Main()
+    {
+        var s = new[] { string.Empty };
+        s[0].ToString();
+        var t = new[] { string.Empty, null };
+        t[0].ToString();
+        var u = new[] { 1, null };
+        u[0].ToString();
+        var v = new[] { null, (int?)2 };
+        v[0].ToString();
```

Comment:
As a side-note, it is interesting that we warn about possible dereference on string?, but not on accessing an int? that may not have a value. 
Should we raise this with LDM as a possible additional warning, for consistency?